packages/components/src/toggle-group-control/toggle-group-control/utils.ts

```diff
@@ -21,16 +21,21 @@ type ValueProp = ToggleGroupControlProps[ 'value' ];
 export function useComputeControlledOrUncontrolledValue(
 	valueProp: ValueProp
 ): { value: ValueProp; defaultValue: ValueProp } {
+	const isInitialRender = useRef( true );
 	const prevValueProp = usePrevious( valueProp );
 	const prevIsControlled = useRef( false );
 
+	useEffect( () => {
+		if ( isInitialRender.current ) {
+			isInitialRender.current = false;
+		}
+	}, [] );
+
 	// Assume the component is being used in controlled mode on the first re-render
 	// that has a different `valueProp` from the previous render.
 	const isControlled =
 		prevIsControlled.current ||
-		( prevValueProp !== undefined &&
-			valueProp !== undefined &&
-			prevValueProp !== valueProp );
```

Comment:
In essence, all we want to do here is check that the current value is different from the previous value.

We can't use `prevValueProp !== undefined` or `valueProp !== undefined` as checks, because it's possible in the first few renders of a controlled usage that either the previous value or the current value is undefined.

Reply:
This also makes sense.

The intention behind the previous check was:

_"the `value` prop was updated from a defined value to another defined value"_

But as you say, I believe that those checks can be removed, since they were kind of redundant: the `prevIsControlled.current` flag makes sure that there is no going back to uncontrolled once controlled, even if `value` suddenly becomes `undefined` again.